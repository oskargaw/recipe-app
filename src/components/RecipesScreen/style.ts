import styled from "styled-components";

export const MainContainer = styled.div`
  width: 100vw;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-family: sans-serif;
`;

export const RecipesContainer = styled.div`
  padding: 2rem 2.5rem;
  box-shadow: 2px 0px 21px -7px rgba(66, 68, 90, 1);
  border-radius: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  @media (min-width: 768px) {
    padding: 3rem 5rem;
  }
`;
