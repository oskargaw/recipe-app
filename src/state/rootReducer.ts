import { combineReducers } from "redux";
import { recipesReducer } from "./recipes/recipes.reducer";

const rootReducer = combineReducers({
  recipes: recipesReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
