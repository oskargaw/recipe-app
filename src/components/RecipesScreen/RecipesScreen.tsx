import { AnimatePresence } from "framer-motion";
import { useState } from "react";
import { useSelector } from "react-redux";
import { getRecipes } from "../../state/recipes/recipes.selectors";
import { Button } from "../Button";
import { RecipeList } from "../RecipeList";
import { RecipeModal } from "../RecipeModal";
import { MainContainer, RecipesContainer } from "./style";

export const RecipesScreen = () => {
  const [recipeModalOpen, setRecipeModalOpen] = useState(false);

  const recipes = useSelector(getRecipes);

  const closeRecipeModal = () => setRecipeModalOpen(false);
  const openRecipeModal = () => setRecipeModalOpen(true);

  return (
    <MainContainer>
      <RecipesContainer>
        {recipes.length === 0 ? (
          <h1>No recipes here!</h1>
        ) : (
          <RecipeList recipes={recipes} />
        )}

        <Button
          variant="primary"
          size="md"
          onClick={() =>
            recipeModalOpen ? closeRecipeModal() : openRecipeModal()
          }
        >
          Add recipe
        </Button>
      </RecipesContainer>

      <AnimatePresence
        initial={false}
        exitBeforeEnter={true}
        onExitComplete={() => null}
      >
        {recipeModalOpen && <RecipeModal handleClose={closeRecipeModal} />}
      </AnimatePresence>
    </MainContainer>
  );
};
