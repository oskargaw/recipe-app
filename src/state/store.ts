import { createStore } from "redux";
import { persistReducer, persistStore } from "redux-persist"; // imports from redux-persist
import storage from "redux-persist/lib/storage";
import rootReducer from "./rootReducer";

const persistConfig = {
  key: "root",
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(
  persistedReducer,
  (window as any).__REDUX_DEVTOOLS_EXTENSION__ &&
    (window as any).__REDUX_DEVTOOLS_EXTENSION__()
);

const persistor = persistStore(store as any);

export { store, persistor };
