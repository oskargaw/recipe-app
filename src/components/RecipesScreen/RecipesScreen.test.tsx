import { fireEvent, render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { RootState } from "../../state/rootReducer";
import { RecipesScreen } from "./RecipesScreen";

describe("<RecipeListItem />", () => {
  const mockStore = configureStore([]);
  const initialState: RootState = {
    recipes: {
      recipeList: [],
    },
  };
  const store = mockStore(initialState);

  it("should render correctly", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <RecipesScreen />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should open recipe modal after clicking add recipe button", () => {
    render(
      <Provider store={store}>
        <RecipesScreen />
      </Provider>
    );

    const modalHeader = screen.queryByText("Edit recipe");
    expect(modalHeader).not.toBeInTheDocument();

    const addRecipeButton = screen.getByText("Add recipe");

    fireEvent.click(addRecipeButton);

    const modalHeaderTitle = screen.getAllByText("Add recipe")[0];
    expect(modalHeaderTitle).toBeInTheDocument();
  });
});
