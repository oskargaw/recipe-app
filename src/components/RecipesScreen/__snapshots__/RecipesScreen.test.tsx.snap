// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeListItem /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-hBUSln czpzSA"
  >
    <div
      class="sc-fotOHu fOFNvG"
    >
      <h1>
        No recipes here!
      </h1>
      <button
        class="sc-bdvvtL fEPyxY"
      >
        Add recipe
      </button>
    </div>
  </div>
</DocumentFragment>
`;
